addons/mail/static/src/messaging/entity/notification_group/notification_group.js

```diff
@@ -0,0 +1,109 @@
+odoo.define('mail.messaging.entity.NotificationGroup', function (require) {
+'use strict';
+
+const { registerNewEntity } = require('mail.messaging.entityCore');
+const { attr, one2many } = require('mail.messaging.EntityField');
+
+function NotificationGroupFactory({ Entity }) {
+
+    class NotificationGroup extends Entity {
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * Opens the view that displays either the single record of the group or
+         * all the records in the group.
+         */
+        openDocuments() {
+            if (this.res_id) {
+                if (this.model === 'mail.channel') {
+                    const channel = this.env.entities.Thread.insert({
+                        id: this.res_id,
+                        model: this.model,
+                    });
+                    channel.open();
+                } else {
+                    this.env.messaging.openDocument({
+                        id: this.res_id,
+                        model: this.model,
+                    });
+                }
+            } else {
+                this._openDocuments();
```

Comment:
it's weird to have same name method that just differs from visibility in their name, but private method actually is a special case of opening documents. I would just dump code of `_openDocuments` here

Reply:
Private method is used for override. The "other case" of the public method doesn't need to be overridden.

Reply:
Basically the public method should be called `openDocumentOrDocuments`, and only the plural one need the override. Not sure that's a good name though.

Reply:
BTW I think the naming "document" was a mistake, and we should have used "record".
From the look of it, it either opens one record or a list of records (the latter is solely based on `model`).

I think we could simply override `openRecords()` (public method) and the override is almost the same (just add `!this.res_id` in conditional)